qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -296,7 +296,9 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
 
             # sample from [0, power) if specified
             if self._sample_from_iterations:
-                power = algorithm_globals.random.integers(power)
+                with warnings.catch_warnings():
+                    warnings.filterwarnings("ignore", category=DeprecationWarning)
+                    power = algorithm_globals.random.integers(power)
```

Comment:
We don't need to (and mostly shouldn't) do this in deprecated library code.  It's ok for the deprecated code to call other deprecated code, and the `stacklevel` should be set on `algorithms_global` correctly such that this file gets the "blame", which makes the warning invisible to users.